Gulpfile.mjs

```diff
@@ -460,20 +516,90 @@ function copyDts(packages) {
     .pipe(gulp.dest(monorepoRoot));
 }
 
-const libBundles = [
-  "packages/babel-parser",
-  "packages/babel-plugin-proposal-object-rest-spread",
-  "packages/babel-plugin-proposal-optional-chaining",
-  "packages/babel-preset-react",
-  "packages/babel-preset-typescript",
-  "packages/babel-helper-member-expression-to-functions",
-  "packages/babel-plugin-bugfix-v8-spread-parameters-in-optional-chaining",
-  "packages/babel-plugin-bugfix-safari-id-destructuring-collision-in-function-expression",
-].map(src => ({
-  src,
-  format: "cjs",
-  dest: "lib",
-}));
+function* libBundlesIterator() {
+  for (const packageDir of ["packages", "codemods"]) {
+    for (const dir of fs.readdirSync(new URL(packageDir, import.meta.url))) {
+      if (
+        // @rollup/plugin-commonjs will mess up with babel-helper-fixtures
+        dir === "babel-helper-fixtures" ||
+        // babel-standalone is handled by rollup-babel-standalone task
+        dir === "babel-standalone" ||
+        // todo: Rollup hangs on allowHashBang: true with babel-cli/src/babel/index.ts hashbang
+        dir === "babel-cli" ||
+        // todo: @rollup/node-resolve 'browsers' option does not work when package.json contains `exports`
+        // https://github.com/rollup/plugins/tree/master/packages/node-resolve#browser
+        dir === "babel-register" ||
+        dir === "babel-core" ||
+        dir === "babel-plugin-transform-runtime" ||
+        // @babel/node invokes internal lib/_babel-node.js
+        dir === "babel-node" ||
+        // todo: test/helpers/define-helper requires internal lib/helpers access
+        dir === "babel-helpers"
+      ) {
+        continue;
+      }
+
+      let pkgJSON;
+      try {
+        pkgJSON = JSON.parse(
+          fs.readFileSync(
+            new URL(`${packageDir}/${dir}/package.json`, import.meta.url)
+          )
+        );
+      } catch (err) {
+        if (err.code !== "ENOENT" && err.code !== "ENOTDIR") throw err;
+      }
+      if (pkgJSON === undefined) continue;
+      const src = packageDir + "/" + dir;
+      if (pkgJSON.main) {
```

Comment:
We may extend support to `pkgJSON.exports` in the future. However, bundling a package with multiple exports could be tricky: Do we want to bundle multiple entry thus minimizing internal module imports? Or do we want to bundle modules in topological order thus minimizing the artifact size? 

Since few packages supports multiple exports, I think supporting `main` is good enough.